### PR TITLE
chore: update mergify config

### LINKED ISSUE #143: Set up mergify in cosmos-sdk fork

### PR DESCRIPTION
**Pull request by alexanderbez**
## Description

Closes: #143

Updated the automerge and backporting configurations

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -4,30 +4,41 @@ queue_rules:
...
+  - name: backport patches to v0.45.0x-osmo-v7 branch
+    conditions:
+      - base=osmosis-main
+      - label=A:backport/v0.45.0x-osmo-v7
+    actions:
+      backport:
+        branches:
+          - v0.45.0x-osmo-v7
+
+  - name: backport patches to v0.45.0x-osmo-v8 branch
```
Comment (p0mvn) [inline .mergify.yml:56]: q: should we be using this tag before v9 is released? My understanding is that we would start using `v0.45.0x-osmo-v8` once v8 is released

File: .mergify.yml [hunk 2]
```diff
@@ -4,30 +4,41 @@ queue_rules:
...
         name: default
         method: squash
         commit_message_template: |
-           {{ title }} (#{{ number }})
-           {{ body }}
-  - name: backport patches to v0.42.x branch
+          {{ title }} (#{{ number }})
+          {{ body }}
+
+  - name: backport patches to release/v0.42.x branch
```
Comment (p0mvn) [inline .mergify.yml:20]: We also have:

- `v0.44.3x-osmo-v5` for v5/v6
- `v0.42.9-osmo-v4` for v4

Should we include backport labels for them as well? Would be helpful with: https://github.com/osmosis-labs/osmosis/issues/1044

What are the branch names for pre-v4? CC: @ValarDragon

**Review by p0mvn — APPROVED** [review]
LGTM! Thanks for adding these configs

Had 2 questions about whether we want to backport to earlier branches and the right use of `v0.45.0x-osmo-v8`

File: .mergify.yml (hunk 2 above)
Comment (alexanderbez) [reply .mergify.yml:20]: Yes, I will add them.

File: .mergify.yml (hunk 1 above)
Comment (alexanderbez) [reply .mergify.yml:56]: This already existed in the config so I just updated it here. Also, `v0.45.0x-osmo-v8` already exists.

**Pull request merged**